### PR TITLE
Elb tagging

### PR DESCRIPTION
**Pull request by cdaniluk**

### CODE REVIEW

File: elb.tf [hunk 1]
```diff
@@ -27,14 +27,13 @@ resource "aws_security_group_rule" "elb_egress" {
 resource "aws_security_group_rule" "elb_ingress" {
   count = length(var.elb_allowed_cidr_blocks) > 0 ? 1 : 0
 
-  cidr_blocks              = var.elb_allowed_cidr_blocks #tfsec:ignore:AWS007
-  description              = "Allow traffic from the allowed ranges"
-  from_port                = 443
-  protocol                 = "tcp"
-  security_group_id        = aws_security_group.elb.id
-  source_security_group_id = aws_security_group.this.id
```
Comment (smiller171) [inline elb.tf:35]: Why are you removing the source security group here?

File: elb.tf [hunk 2]
```diff
@@ -62,10 +61,12 @@ resource "aws_lb_listener" "this" {
 
 resource "aws_lb_target_group" "this" {
   name_prefix = substr(var.name, 0, 6)
-  port        = "80"
-  protocol    = "HTTP"
-  tags        = var.tags
-  vpc_id      = var.vpc_id
+  # Nexus has a bad time if two instances are running at once, so the deregistration delay needs to be short
```
Comment (smiller171) [inline elb.tf:64]: How dangerous is it to set the delay to 0 and just terminate the old instance?

File: main.tf [hunk 3]
```diff
@@ -20,7 +20,7 @@ data "template_cloudinit_config" "this" {
 resource "aws_autoscaling_group" "this" {
   name_prefix               = var.name
   desired_capacity          = var.asg_desired_capacity
-  health_check_grace_period = 300
+  health_check_grace_period = 600
```
Comment (smiller171) [inline main.tf:23]: 10 minutes of grace on the health check is a long damn time. Does it really take that long to start up?

**Review by smiller171 — COMMENTED** [review]
Nothing wrong perse, just not sure why you're making these particular changes.

Also you called the PR "Elb tagging" but I don't see anything related to tagging in these changes

**Review by sblack4 — COMMENTED** [review]
I think Scott has good questions

File: elb.tf (hunk 1 above)
Comment (cdaniluk) [reply elb.tf:35]: you can't mix `source_security_group_id` in the same `aws_security_group_rule` with `cidr_blocks`. This was the primary issue I was fixing in this PR. I honestly didn't realize `elb-tagging` had been merged already and reopened the branch applying this fix. Too many things going on at once :(

File: elb.tf (hunk 2 above)
Comment (cdaniluk) [reply elb.tf:64]: I'm not sure. I thought about it but I figured at least this way there's no chance Nexus would still have a lock on the running repo by the time cloud-init finished on the new host, but there would be _some_ time for the Java process to shut down naturally-ish.

File: main.tf (hunk 3 above)
Comment (cdaniluk) [reply main.tf:23]: It takes more than 5 minutes sometimes as it does pre-flight checks on every repo.

**Review by sblack4 — APPROVED** [review]
:shipit:

**Pull request merged**